src/dynsight/_internal/trajectory/trajectory.py

```diff
@@ -61,131 +84,206 @@ def get_onion(self, delta_t: int) -> OnionInsight:
         onion_clust.fit(reshaped_data)
 
         return OnionInsight(
-            dataset=self.dataset,
-            r_cut=self.r_cut,
             labels=onion_clust.labels_,
-            delta_t=delta_t,
             state_list=onion_clust.state_list_,
             reshaped_data=reshaped_data,
+            meta={"delta_t": delta_t},
         )
 
-    def save(self, file_name: str) -> None:
-        file_path = Path(f"{file_name}.pkl")
-        with Path.open(file_path, "wb") as file:
-            pickle.dump(self, file)
 
-
-@dataclass
-class ClusterInsight(Insight):
+@dataclass(frozen=True)
+class ClusterInsight:
     """Contains a clustering analysis."""
 
     labels: NDArray[np.int64]
 
+    def dump_to_json(self, file_path: Path) -> None:
+        """Save the ClusterInsight object as .json file."""
+        data = self.__dict__
+        data["labels"] = data["labels"].tolist()
+        with file_path.open("w") as file:
+            json.dump(data, file, indent=4)
+
+    @classmethod
+    def load_from_json(cls, file_path: Path) -> ClusterInsight:
+        """Load the ClusterInsight object from .json file."""
+        with file_path.open("r") as file:
+            data = json.load(file)
+        return cls(labels=np.array(data.get("labels")))
```

Comment:
Here, and below, is it worth including a raise error for if `labels` not in the json?